app-agent/src/exec_process.rs

```diff
@@ -202,48 +138,88 @@ fn handle_not_found(external_command: String, args: Vec<String>) -> String {
     }
     match best_element {
         Some((element, distance)) => {
+            let argument_list = args.iter().map(|arg| {
+                if arg.contains(' ') {
+                   format!("\"{}\"", arg)
+               } else {
+                   arg.to_string()
+               }
+           }).collect::<Vec<_>>().join(" ");
             if distance == 0 {
                 log::info!(
                     "💡 Hint: A file named '{}' exists in the current directory. Prepend ./ to execute it.",
                     element
                 );
-                log::info!(
-                    "Example: {} exec ./{} {}",
-                    app_path,
-                    element,
-                    args.iter()
-                        .map(|arg| {
-                            if arg.contains(' ') {
-                                format!("\"{}\"", arg)
-                            } else {
-                                arg.to_string()
-                            }
-                        })
-                        .collect::<Vec<_>>()
-                        .join(" ")
-                );
+                log::info!("Example: {} exec ./{} {}", app_path, element, argument_list);
             } else {
-                log::info!(
-                    "💡 Hint: Did you mean ./{} {}",
-                    element,
-                    args.iter()
-                        .map(|arg| {
-                            if arg.contains(' ') {
-                                format!("\"{}\"", arg)
-                            } else {
-                                arg.to_string()
-                            }
-                        })
-                        .collect::<Vec<_>>()
-                        .join(" ")
-                );
+                log::info!("💡 Hint: Did you mean ./{} {}", element, argument_list);
             }
         }
         None => {
-            log::warn!("💡 Hint: No matching file exists in the current directory. Prepend ./ to execute it.");
+            log::warn!("💡 Hint: No matching file exists in the current directory. Please try again we a different one.");
+        }
+    }
+    "Sorry but the file was not found".to_string()
+}
+
+fn check_missing_permissions(current_permissions: u32, required_permissions: u32) -> u32 {
```

Comment:
Is the returned number actually used as a number or only as a "rwx" string? If the latter, that string could be constructed directly in this function.

Reply:
@AngeCyp